File: algorithms/Convert Binary Number in a Linked List to Integer/Convert Binary Number in a Linked List to Integer.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def getDecimalValue(self, head: ListNode) -> int:
        res = 0
        po = 0
        stack = []
        node = head
        while node:
            stack.append(node.val)
            node = node.next
        res = 0
        for i in reversed(stack):
            res += i*(2**po)
            po += 1
        return res

File: scripts/algorithms/C/Convert Binary Number in a Linked List to Integer/Convert Binary Number in a Linked List to Integer.rs
// Runtime: 1 ms (Top 70.8%) | Memory: 1.96 MB (Top 87.5%)

impl Solution {
    pub fn get_decimal_value(head: Option<Box<ListNode>>) -> i32 {
        let mut cur_node = &head;
        let mut v = Vec::new();
        while let Some(node) = cur_node {
            v.push(node.val);
            cur_node = &node.next;
        }

        v.iter()
            .rev()
            .enumerate()
            .fold(0, |acc, (ind, val)| acc + *val * (1 << ind))
    }
}